src/Nethermind/Nethermind.Provide.Vault/VaultSealingHelper.cs

```diff
@@ -0,0 +1,112 @@
+        {
+            _vaultKeyStoreFacade = vaultKeyStoreFacade ?? throw new ArgumentNullException(nameof(vaultKeyStoreFacade));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+        public void Seal()
+        {
+            try
+            {
+                var sealResult = SealingUnsealingMethod("seal");
+                if (sealResult.Success)
+                {
+                    if (_logger.IsInfo) _logger.Info($"The vault sealing was successful.");
+                }
+                else
+                {
+                    if (_logger.IsError) _logger.Error($"The vault sealing was failed. {sealResult.Error}");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"Failed to seal vault", ex);
+            }
+        }
+
+        public void Unseal()
+        {
+            try
+            {
+                var unsealResult = SealingUnsealingMethod("unseal");
+                if (unsealResult.Success)
+                {
+                    if (_logger.IsInfo) _logger.Info($"The vault unsealing was successful.");
+                }
+                else
+                {
+                    if (_logger.IsError) _logger.Error($"The vault unsealing was failed. {unsealResult.Error}");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"Failed to unseal vault", ex);
+            }
+        }
+
+        class SealingUnsealingRequest
+        {
+            public string key { get; set; }
+        }
+        public (bool Success, string Error) SealingUnsealingMethod(string methodName)
+        {
+            var request = new SealingUnsealingRequest()
+            {
+                key = _vaultKeyStoreFacade.GetKey()
+            };
+
+            // with a new version of the Provide Nuget package we should remove httpClient call and use the provide.Unseal method
+            using (var httpClient = new HttpClient())
+            {
+                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("bearer", _config.Token);
+                StringContent content = new StringContent(JsonConvert.SerializeObject(request), Encoding.UTF8, "application/json");
+                var url = ConstructUrl(methodName);
+                var response = httpClient.PostAsync(url, content).Result;
+                var responseStr = $"Status code: {response.StatusCode}, ResaonPhrase: {response.ReasonPhrase} Content: {response.Content.ReadAsStringAsync().Result}";
+                return (response.IsSuccessStatusCode, responseStr);
+            }
+        }
+
+        private string ConstructUrl(string methodName)
+        {
+            return _config.Scheme + "://" + _config.Host + "/" + _config.Path + $"/{methodName}";
```

Comment:
UriBuilder

Reply:
> UriBuilder

We don't expose port in our config. We will get "http://[localhost:8082]/api/v1/seal". Furthermore, the problem won't exist when the Provide add that method to Nuget.

Reply:
UriBuilder